node/pkg/watchers/sui/watcher.go

```diff
@@ -503,25 +443,146 @@ func (e *Watcher) Run(ctx context.Context) error {
+				done = true
+				break
+			} else {
+				retVal.Data = append(retVal.Data, datum)
+			}
+		}
+		if (!done) && res.Result.HasNextPage {
+			nextCursor.TxDigest = res.Result.NextCursor.TxDigest
+			nextCursor.EventSeq = res.Result.NextCursor.EventSeq
+		} else {
+			done = true
+		}
+	}
+
+	retVal.Data = reverseArray(retVal.Data)
+	return retVal, nil
+}
+
+func (e *Watcher) getLatestCheckpointSN(logger *zap.Logger) (int64, error) {
+	resp, err := http.Post(e.suiRPC, "application/json", strings.NewReader(`{"jsonrpc":"2.0", "id": 1, "method": "sui_getLatestCheckpointSequenceNumber", "params": []}`)) //nolint:noctx // TODO FIXME we should propagate context with Deadline here.
+	if err != nil {
+		logger.Error("sui_getLatestCheckpointSequenceNumber failed", zap.Error(err))
+		p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDSui, 1)
+		return 0, fmt.Errorf("sui_getLatestCheckpointSequenceNumber failed to post: %w", err)
+	}
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		logger.Error("sui_getLatestCheckpointSequenceNumber failed", zap.Error(err))
+		p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDSui, 1)
+		return 0, fmt.Errorf("sui_getLatestCheckpointSequenceNumber failed to read: %w", err)
+	}
+	resp.Body.Close()
+	logger.Debug("Body before unmarshalling", zap.String("body", string(body)))
+	var res SuiCheckpointSN
+	err = json.Unmarshal(body, &res)
+	if err != nil {
+		logger.Error("unmarshal failed into uint64", zap.String("body", string(body)), zap.Error(err))
+		p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDSui, 1)
+		return 0, fmt.Errorf("sui_getLatestCheckpointSequenceNumber failed to unmarshal body: %s, error: %w", string(body), err)
+	}
+	height, pErr := strconv.ParseInt(res.Result, 0, 64)
+	if pErr != nil {
+		logger.Error("Failed to ParseInt")
+	} else {
+		currentSuiHeight.Set(float64(height))
+		logger.Debug("sui_getLatestCheckpointSequenceNumber", zap.String("result", res.Result))
+
+		p2p.DefaultRegistry.SetNetworkStats(vaa.ChainIDSui, &gossipv1.Heartbeat_Network{
+			Height:          height,
+			ContractAddress: e.suiMoveEventType,
+		})
+	}
+	return height, nil
+}
 
-	// When the scheme is empty/nil or when the Host is empty but a scheme is set
-	if u == nil || u.Scheme == "" || (u.Scheme != "" && u.Host == "") {
-		logger.Warn(fmt.Sprintf("DEPRECATED: Prefix --suiWS address with the url scheme e.g.: ws://%s or wss://%s", e.suiWS, e.suiWS))
-		u = &url.URL{Scheme: "ws", Host: e.suiWS}
-	} else if u.Scheme != "ws" && u.Scheme != "wss" {
-		return fmt.Errorf("invalid url scheme specified for --suiWS, try ws:// or wss://: %s", e.suiWS)
+func (e *Watcher) getCheckpointForEvent(event SuiResult) (int64, error) {
+	retVal := int64(0)
+	reader := strings.NewReader(fmt.Sprintf(`{"jsonrpc":"2.0", "id": 1, "method": "sui_getTransactionBlock", "params": [ "%s" ]}`,
+		*event.ID.TxDigest))
+	resp, err := http.Post(e.suiRPC, "application/json", reader) //nolint:noctx // TODO FIXME we should propagate context with Deadline here.
```

Comment:
@evan-gray  Should this be fixed?

Reply:
This was how Toby left things.  Fix here or in a different PR?

Reply:
I realized that too after my comment. I guess it can be fixed in the future, though I don't know if there's an increased impact with the increased number of rpc calls.

Reply:
If I do a `context.WithTimeout()`, what should I set the timeout value to?
@evan-gray ☝️

Reply:
I'm not evan-gray, but I play one on TV. 😁 

Maybe five or ten seconds?

Reply:
I made it a variable.